auth/auth_test.go

```diff
@@ -125,3 +134,62 @@ func TestSignatureTxMiddlewareMultipleTxSameBlock(t *testing.T) {
 	NonceTxPostNonceMiddleware(state, nonceTxBytes, loomchain.TxHandlerResult{}, nil)
 
 }
+
+func TestRevertedTxNonceMiddleware(t *testing.T) {
+	pubkey, _, err := ed25519.GenerateKey(nil)
+	if err != nil {
+		panic(err)
+	}
+
+	nonceTxBytes, err := proto.Marshal(&NonceTx{
+		Inner:    []byte{},
+		Sequence: 1,
+	})
+	require.NoError(t, err)
+
+	nonceTxBytes2, err := proto.Marshal(&NonceTx{
+		Inner:    []byte{},
+		Sequence: 2,
+	})
+	require.NoError(t, err)
+
+	origin := loom.Address{
+		ChainID: "default",
+		Local:   loom.LocalAddressFromPublicKey(pubkey),
+	}
+
+	ctx := context.WithValue(context.Background(), ContextKeyOrigin, origin)
+	ctx = context.WithValue(ctx, ContextKeyCheckTx, true)
+	kvStore := store.NewMemStore()
+	storeTx := store.WrapAtomic(kvStore).BeginTx()
+	state := loomchain.NewStoreState(ctx, storeTx, abci.Header{Height: 27}, nil, nil)
+	state.SetFeature(loomchain.IncrementNonceFailedTxFeature, true)
+
+	// Nonce is 0
+	currentNonce := Nonce(state, origin)
+	require.Equal(t, uint64(0), currentNonce)
+
+	// Send a successful tx
+	_, err = NonceTxHandler.Nonce(state, kvStore, nonceTxBytes,
+		func(state loomchain.State, txBytes []byte, isCheckTx bool) (loomchain.TxHandlerResult, error) {
+			return loomchain.TxHandlerResult{}, nil
+		}, false,
+	)
+	require.Nil(t, err)
+	NonceTxPostNonceMiddleware(state, nonceTxBytes, loomchain.TxHandlerResult{}, nil)
+	storeTx.Commit()
+	storeTx.Rollback()
+
+	// Send a failed tx, nonce should increase even though the transaction is reverted
+	_, err = NonceTxHandler.Nonce(state, kvStore, nonceTxBytes2,
+		func(state loomchain.State, txBytes []byte, isCheckTx bool) (loomchain.TxHandlerResult, error) {
+			return loomchain.TxHandlerResult{}, errors.New("EVM transaction reverted")
+		}, false,
+	)
+	require.Error(t, err)
+	NonceTxPostNonceMiddleware(state, nonceTxBytes, loomchain.TxHandlerResult{}, nil)
```

Comment:
PostCommit middleware doesn't run on failed txs does it?

Reply:
It run on `CheckTx` but it does not run on `DeliverTx`